src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/FlowModelInitializer.cs

```diff
@@ -1,36 +1,116 @@
 
 namespace Dfc.CourseDirectory.WebV2.Features.NewApprenticeshipProvider
 {
     public class FlowModelInitializer
     {
         private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ISqlQueryDispatcher _sqlDbQueryDispatcher;
+        private readonly IStandardsAndFrameworksCache _iStandardandFromworkCache;
 
-        public FlowModelInitializer(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        public FlowModelInitializer(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher, ISqlQueryDispatcher sqlDbQueryDispatcher, IStandardsAndFrameworksCache iStandardandFromworkCache)
         {
             _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _sqlDbQueryDispatcher = sqlDbQueryDispatcher;
+            _iStandardandFromworkCache = iStandardandFromworkCache;
         }
 
         public async Task<FlowModel> Initialize(Guid providerId)
         {
+            var model = new FlowModel();
             var provider = await _cosmosDbQueryDispatcher.ExecuteQuery(
                 new GetProviderById()
                 {
                     ProviderId = providerId
                 });
 
-            var model = new FlowModel();
 
+            var submission = await _sqlDbQueryDispatcher.ExecuteQuery(new GetLatestApprenticeshipQASubmissionForProvider()
+            {
+                ProviderId = providerId
+            });
+
+            var existingSubmission = submission.Match(_ => null,
+                                                      match => match);
+
+            if (existingSubmission != null)
+            {
+                var apprenticeship = existingSubmission.Apprenticeships.Single();
+                var apprenticeshipId = apprenticeship.ApprenticeshipId;
+                var cosmosApprenticeship = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeshipsByIds() { ApprenticeshipIds = new Guid[] { apprenticeshipId } });
+
+                if (cosmosApprenticeship.ContainsKey(apprenticeshipId))
+                {
+                    Standard standard = default;
+                    Framework framework = default;
+
+                    if (cosmosApprenticeship[apprenticeshipId].StandardCode.HasValue)
+                        standard = await _iStandardandFromworkCache.GetStandard(cosmosApprenticeship[apprenticeshipId].StandardCode.Value, cosmosApprenticeship[apprenticeshipId].Version.Value);
+
+                    if (cosmosApprenticeship[apprenticeshipId].FrameworkCode.HasValue)
+                        framework = await _iStandardandFromworkCache.GetFramework(cosmosApprenticeship[apprenticeshipId].FrameworkCode.Value, cosmosApprenticeship[apprenticeshipId].ProgType.Value, cosmosApprenticeship[apprenticeshipId].PathwayCode.Value);
+
+                    model.ApprenticeshipWebsite = cosmosApprenticeship[apprenticeshipId].ContactWebsite;
+                    model.ApprenticeshipStandardOrFramework = cosmosApprenticeship[apprenticeshipId].StandardCode.HasValue ?
```

Comment:
I'd move these assignments inside the `if` blocks above

Reply:
done